src/Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -179,6 +183,60 @@ private async Task SetKernel(IKernelCommand command, KernelInvocationContext con
             }
         }
 
+        private async Task HandleLinePositionLanguageServiceRequests(
+            LanguageServiceCommandBase languageServiceCommand,
+            KernelInvocationContext context,
+            KernelPipelineContinuation continueOnCurrentPipeline)
+        {
+            var tree = SubmissionParser.Parse(languageServiceCommand.Code, languageServiceCommand.TargetKernelName);
+            var nodes = tree.GetRoot().ChildNodes.ToArray();
+            if (nodes.Length == 1)
+            {
+                // no splitting happened, continue like normal
+                await continueOnCurrentPipeline(languageServiceCommand, context);
```

Comment:
All of this logic is pretty awkward and I'm super hesitant to duplicate it, especially since we already know this also needs to differ between completions (i.e. magic commands need to broadcast to all subkernels) vs hover text.

We should discuss how we can simplify.